### PR TITLE
add load caffe model doc, remove useless parameter, add python api

### PR DESCRIPTION
**Pull request by SeaOfOcean**
## What changes were proposed in this pull request?

add load caffe model doc, remove useless parameter, add python api

## How was this patch tested?

unit test

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/1260

### CODE REVIEW

File: docs/docs/ProgrammingGuide/caffe-support.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+## **Load and Save a Caffe Model**
+
+If you have a pretrained caffe model(model definition prototxt and model binary file), you can load it into BigDL model.
```
Comment (wzhongyuan) [inline docs/docs/ProgrammingGuide/caffe-support.md]: file -> files ?  load it into BigDL Model => load it as BigDL model ?

Comment (wzhongyuan) [inline docs/docs/ProgrammingGuide/caffe-support.md]: or convert to BigDL model

File: docs/docs/ProgrammingGuide/caffe-support.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+* Scala Example
+```scala
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric.NumericFloat
+bigdlModel.saveCaffe(prototxtPath, modelPath, useV2 = true, overwrite = false)
+```
+
+* Python Example
+```python
+bigdl_model.save_caffe(prototxt_path, model_path, use_v2 = True, overwrite = False)
+```
```
Comment (wzhongyuan) [inline docs/docs/ProgrammingGuide/caffe-support.md]: we may want to light a bit about the meanings of parameters like use_v2 , overrite, etc.

File: docs/docs/ProgrammingGuide/caffe-support.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
+## **Load and Save a Caffe Model**
```
Comment (shane-huang) [inline docs/docs/ProgrammingGuide/caffe-support.md]: remove this title, as we can't have 2 levels of headers in one MD.

File: docs/docs/ProgrammingGuide/caffe-support.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+If you have a pretrained caffe model(model definition prototxt and model binary file), you can load it into BigDL model.
+You can also convert a BigDL model to caffe model.
+
+### Load Caffe Model
```
Comment (shane-huang) [inline docs/docs/ProgrammingGuide/caffe-support.md]: change this to ## \*\*Load Caffe Model**, so as the following subsection titles.

**Comment by SeaOfOcean** [comment]
pr validate pass

**Comment by wzhongyuan** [comment]
LGTM

**Pull request merged**